src/mumble/AudioStats.cpp

```diff
@@ -216,34 +216,28 @@ void AudioNoiseWidget::paintEvent(QPaintEvent *) {
 	paint.fillRect(rect(), pal.color(QPalette::Window));
 
 	AudioInputPtr ai = Global::get().ai;
-	if (!ai.get() || !ai->sppPreprocess)
+	if (!ai.get() || !ai->m_preprocessor)
 		return;
 
 	QPolygonF poly;
 
 	ai->qmSpeex.lock();
 
-	spx_int32_t ps_size = 0;
-	speex_preprocess_ctl(ai->sppPreprocess, SPEEX_PREPROCESS_GET_PSD_SIZE, &ps_size);
-
-	static std::vector< spx_int32_t > noise;
-	noise.resize(static_cast< std::size_t >(ps_size));
-	static std::vector< spx_int32_t > ps;
-	ps.resize(static_cast< std::size_t >(ps_size));
-
-	speex_preprocess_ctl(ai->sppPreprocess, SPEEX_PREPROCESS_GET_PSD, ps.data());
-	speex_preprocess_ctl(ai->sppPreprocess, SPEEX_PREPROCESS_GET_NOISE_PSD, noise.data());
+	const std::vector< std::int32_t > ps    = ai->m_preprocessor.getPSD();
+	const std::vector< std::int32_t > noise = ai->m_preprocessor.getNoisePSD();
```

Comment:
For better refactor-ability there should probably a typedef like `psd_t` in the processor class that can be used here.